piet-direct2d/src/lib.rs

```diff
@@ -456,21 +457,43 @@ impl<'a> RenderContext for D2DRenderContext<'a> {
     fn capture_image_area(&mut self, rect: impl Into<Rect>) -> Result<Self::Image, Error> {
         let r = rect.into();
 
+        let (dpi_scale, _) = self.rt.get_dpi_scale();
+        let dpi_scale = dpi_scale as f64;
+
+        let transform_matrix = self.rt.get_transform();
+        let affine_transform = matrix3x2f_to_affine(transform_matrix);
+
+        let device_size = Point {
+            x: r.width() * dpi_scale,
+            y: r.height() * dpi_scale,
+        };
+        // TODO: This transformation is untested with the current test pictures
+        let device_size = affine_transform * device_size;
+
+        let device_origin = Point {
+            x: r.x0 * dpi_scale,
+            y: r.y0 * dpi_scale,
+        };
+        // TODO: This transformation is untested with the current test pictures
+        let device_origin = affine_transform * device_origin;
+
         let mut target_bitmap = self.rt.create_blank_bitmap(
-            r.width() as usize,
-            r.height() as usize,
+            device_size.x as usize,
+            device_size.y as usize,
```

Comment:
Now  with `device_size` being a `Size`, this needs to be changed to `device_size.width` and `device_size.height`.